armasec/armasec.py

```diff
@@ -19,27 +22,33 @@ class Armasec:
 
     def __init__(
         self,
-        domain: str,
-        audience: Optional[str] = None,
-        algorithm: str = "RS256",
+        domain_configs: Optional[List[DomainConfig]] = None,
         debug_logger: Optional[Callable[[str], None]] = noop,
         debug_exceptions: bool = False,
+        **kargs,
     ):
         """
         Stores initialization values for the TokenSecurity. All are passed through.
 
         Args:
-            domain:           The OIDC domain where resources are loaded
-            audience:         Optional designation of the token audience.
-            algorithm:        The the algorithm to use for decoding. Defaults to RS256.
+            domain_configs:   List of domain configuration to authenticate the tokens against.
             debug_logger:     A callable, that if provided, will allow debug logging. Should be
                               passed as a logger method like `logger.debug`
             debug_exceptions: If True, raise original exceptions. Should only be used in a testing
                               or debugging context.
+            kargs:             Arguments compatible to instantiate the DomainConfig model.
```

Comment:
I would rather break compatibility and have a new major version than try to support two different ways of instantiating this class. Better to just require one or more `domain_config`.

Reply:
so, my idea is to enable not passing a list of config when you just have a single domain. It doesn't make sense in my mind, that's why I implemented this way.

About the major version: I got no problems with that, it could be IMO.